uniffi_meta/src/lib.rs

```diff
@@ -182,10 +182,72 @@ impl ConstructorMetadata {
     }
 }
 
+// A "sub-type" which represents all the "self" receivers we support.
+// It would ideally be `Type`
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Checksum, Node)]
+pub enum MethodReceiver {
+    Object { module_path: String, name: String },
+    Record { module_path: String, name: String },
+    Enum { module_path: String, name: String },
+}
+
+impl MethodReceiver {
+    pub fn module_path(&self) -> &str {
+        match self {
+            Self::Enum { module_path, .. } => module_path,
+            Self::Record { module_path, .. } => module_path,
+            Self::Object { module_path, .. } => module_path,
+        }
+    }
+
+    pub fn name(&self) -> &str {
+        match self {
+            Self::Enum { name, .. } => name,
+            Self::Record { name, .. } => name,
+            Self::Object { name, .. } => name,
+        }
+    }
+
+    // a bit smelly, but it is what it is.
+    pub fn update_name(&mut self, new_name: String) {
```

Comment:
Is this only used by `uniffi_udl`?  How about making this a function in that crate?